src/transformers/models/mamba2/modeling_mamba2.py

```diff
@@ -0,0 +1,762 @@
+# coding=utf-8
+# Copyright 2024 state-spaces/mamba2 org and HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""PyTorch MAMBA2 model."""
+
+import math
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Tuple, Union
+
+import torch
+import torch.utils.checkpoint
+from torch import nn
+from torch.nn import CrossEntropyLoss
+
+from ...activations import ACT2FN
+from ...modeling_utils import PreTrainedModel
+from ...utils import (
+    ModelOutput,
+    add_code_sample_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    logging,
+)
+from ...utils.import_utils import is_causal_conv1d_available, is_mamba_ssm_available
+from .configuration_mamba2 import Mamba2Config
+
+
+logger = logging.get_logger(__name__)
+
+if is_mamba_ssm_available():
```

Comment:
Should this not be a version check since `mamba_ssm` only provides the necessary functions after v2.0 and there has been a bug on first release that caused NaN losses which has been fixed with [>=2.0.4](https://github.com/state-spaces/mamba/issues/352#issuecomment-2162223186). 

Further [info](https://github.com/state-spaces/mamba/issues/352#issuecomment-2167093395) that might be provided in the docs: There is a restriction on the parameters with `d_model * expand / headdim = multiple of 8`.

Reply:
cc @molbap we need to check version here!